qiime_studio/api/v1.py

```diff
@@ -115,3 +124,74 @@ def api_input_artifacts(plugin_name, workflow_name, input_name):
         if Artifact.load(artifact['path']).type <= input_type:
             input_artifacts.append(artifact)
     return jsonify({'input_artifacts': input_artifacts})
+
+
+@v1.route('/job/<plugin_name>/<workflow_name>', methods=['POST'])
+def execute_workflow(plugin_name, workflow_name):
+    plugin = PLUGIN_MANAGER.plugins[plugin_name]
+    workflow = plugin.workflows[workflow_name]
+
+    request_body = request.get_json()['jobData']
+
+    process_values = {}
+    for key, value in request_body.items():
+        if '-' in key:
+            type_, name = key.split('-')
+            process_values[name] = value
+
+    inputs = {
+        name: process_values[name]
+        for name in workflow.signature.inputs
+    }
+    parameters = {
+        name: process_values[name]
+        for name in workflow.signature.parameters
+    }
+    outputs = {
+        name: process_values[name]
+        for name in workflow.signature.outputs
+    }
+
+    def toggle_completion(process):
+        process_id = id(process)
+        completed_process = process.result()
+        if completed_process.returncode != 0:
+            __JOBS[process_id]['error'] = True
+            __JOBS[process_id]['message'] = \
+                completed_process.stderr.decode('utf-8')
+        __JOBS[process_id]['completed'] = True
+
+    now = '{:%Y-%b-%d %H:%M:%S}'.format(datetime.datetime.now())
+    process = SUBPROCESS_EXECUTOR(workflow, inputs, parameters, outputs)
+    process_id = id(process)
+    __JOBS[process_id] = {
+        'completed': False,
+        'error': False,
+        'message': ''
+    }
+    process.add_done_callback(toggle_completion)
+
+    success = process.running() or process.done()
+    result = {
+        'success': success,
+        'job': {
+            'workflow': workflow_name,
+            'id': process_id,
```

Comment:
I was trying to figure out how to get at that, after @jairideout brought it up the day we were talking about it. The executor returns the `Future` object that is a subprocess, but that process seems to have [no information about the actual QIIME Job Object itself,](https://github.com/qiime2/qiime2/blob/master/qiime/sdk/execution.py#L37:L40) as it's just running the code that was created by the Job. The `id` of the `Future` seemed to be the only thing accessible.